packages/react-admin/src/mui/detail/Edit.js

```diff
@@ -100,18 +100,29 @@ export class Edit extends Component {
     }
 
     updateData(resource = this.props.resource, id = this.props.id) {
-        this.props.crudGetOne(resource, id, this.getBasePath());
+        this.props.crudGetOne({
+            resource,
+            id,
+            basePath: this.getBasePath(),
+        });
     }
 
-    save = (record, redirect) => {
-        this.props.crudUpdate(
-            this.props.resource,
-            this.props.id,
-            record,
-            this.props.data,
-            this.getBasePath(),
-            redirect
-        );
+    save = (record, redirect, dispatch) => {
+        return crudUpdate(
```

Comment:
why not connecting this action ? Connected action creators do not return anything ?

Reply:
The action createor emits a promise so can't be connected as a normal actions object (where each action creator should create a pojo action). That's why it receives dispatch as second parameter. It will dispatch the action internally and return a promise to listen to.